src/biz/metrics.go

```diff
@@ -95,15 +109,31 @@ func (mc *MetricsFetcher) Process(containerID string) (Sample, error) {
 	if err != nil {
 		return metrics, err
 	}
-	// TODO: move logic to skip container without State to Docker specific code.
 	if json.ContainerJSONBase == nil {
 		return metrics, errors.New("empty container inspect result")
 	}
 
+	// TODO: move logic to skip container without State to Docker specific code.
 	if json.State == nil {
 		log.Debug("invalid container %s JSON: missing State", containerID)
 	}
 
+	if mc.exitedContainerTTL != 0 && strings.ToLower(json.State.Status) == "exited" {
+		exitTimestamp, err := time.Parse(time.RFC3339Nano, json.State.FinishedAt)
+		if err != nil {
+			return metrics, fmt.Errorf("invalid finished_at timestamp for exited container %s: %s (%v)",
+				containerID,
+				json.State.FinishedAt,
+				err,
+			)
+		}
+		if time.Now().After(exitTimestamp.Add(mc.exitedContainerTTL)) {
```

Comment:
Really minor nitpick
```suggestion
		if time.Since(exitTimestamp) > mc.exitedContainerTTL {
```

Reply:
Ahhh, this reads much better!